source/codegen/templates/proto_helpers.mako

```diff
@@ -53,6 +53,48 @@ enum ${enum_name} {
 % endfor
 </%def>
 
+## Define enums in the proto per attribute type that uses enum values
+<%def name="define_attribute_values_enums(attribute_enums)">\
+<%
+  config = data["config"]
+  service_class_prefix = config["service_class_prefix"]
+  enums = data["enums"]
+%>\
+% for type_name in attribute_enums:
+<%
+  type_enum_name = service_class_prefix + type_name[2:] + "AttributeValues"
+  enum_value_type_specific_prefix = (f"{service_class_prefix.upper()}_{type_name[2:]}_VAL").upper()
+  values_to_create = { enum_value_type_specific_prefix + "_UNSPECIFIED": 0 }
+  for enum_name in sorted(attribute_enums[type_name]):
+    enum_value_prefix = enum_value_type_specific_prefix + "_" + common_helpers.pascal_to_snake(enum_name)
```

Comment:
Enum name needs to be part of value. In case same enum value is defined in multiple enums in enums.py, not prefixing enum name here would cause conflict. 

Example from dcpower: Two of the values from [PowerSource](https://github.com/doshirohan/grpc-device/blob/main/source/codegen/metadata/nidcpower/enums.py#L521) are also present in `PowerSourceInUse` enum.

I have updated enum names to use your suggestion though and moved `_VAL_` after all the prefixes to keep names close to what C APIs have.